raiden/ui/startup.py

```diff
@@ -0,0 +1,361 @@
+    """Sets the config depending on the environment type"""
+    # interpret the provided string argument
+    if environment_type == Environment.PRODUCTION:
+        # Safe configuration: restrictions for mainnet apply and matrix rooms have to be private
+        config['environment_type'] = Environment.PRODUCTION
+        config['transport']['matrix']['private_rooms'] = True
+    else:
+        config['environment_type'] = Environment.DEVELOPMENT
+
+    print(f'Raiden is running in {environment_type.value.lower()} mode')
+    return environment_type
+
+
+def setup_contracts_or_exit(
+        config: Dict[str, Any],
+        network_id: int,
+) -> Tuple[Dict[str, Any], bool]:
+    """Sets the contract deployment data depending on the network id and environment type
+
+
+    If an invalid combination of network id and environment type is provided, exits
+    the program with an error
+    """
+    environment_type = config['environment_type']
+
+    not_allowed = (  # for now we only disallow mainnet with test configuration
+        network_id == 1 and
+        environment_type == Environment.DEVELOPMENT
+    )
+    if not_allowed:
+        click.secho(
+            f'The chosen network ({ID_TO_NETWORKNAME[network_id]}) is not a testnet, '
+            f'but the "development" environment was selected.\n'
+            f'This is not allowed. Please start again with a safe environment setting '
+            f'(--environment production).',
+            fg='red',
+        )
+        sys.exit(1)
+
+    contracts = dict()
+    contract_addresses_known = False
+    contracts_version = environment_type_to_contracts_version(environment_type)
+
+    config['contracts_path'] = contracts_precompiled_path(contracts_version)
+
+    if network_id in ID_TO_NETWORKNAME and ID_TO_NETWORKNAME[network_id] != 'smoketest':
+        deployment_data = get_contracts_deployment_info(
+            chain_id=network_id,
+            version=contracts_version,
+        )
+
+        contracts = deployment_data['contracts']
+        contract_addresses_known = True
+
+    return contracts, contract_addresses_known
+
+
+def handle_contract_version_mismatch(mismatch_exception: ContractVersionMismatch) -> None:
+    click.secho(
+        f'{str(mismatch_exception)}. Please update your Raiden installation.',
+        fg='red',
+    )
+    sys.exit(1)
+
+
+def handle_contract_no_code(name: str, address: Address) -> None:
+    hex_addr = to_checksum_address(address)
+    click.secho(f'Error: Provided {name} {hex_addr} contract does not contain code', fg='red')
+    sys.exit(1)
+
+
+def handle_contract_wrong_address(name: str, address: Address) -> None:
+    hex_addr = to_checksum_address(address)
+    click.secho(
+        f'Error: Provided address {hex_addr} for {name} contract'
+        ' does not contain expected code.',
+        fg='red',
+    )
+    sys.exit(1)
+
+
+class Proxies(NamedTuple):
```

Comment:
nit: What's the advantage of a `NamedTuple` over a dataclass now that we're on 3.7?

Reply:
It's rather simple. Dataclasses can do a lot more than namedtuples so for more complicated things we can/should use them. But using `NamedTuple` for making a return value of a function more readable than a simple Tuple is I believe much faster, easier to read and more peformant. It's also immutable. Dataclass can also be immutable with `frozen=true` but as I said, this is already more complicated than having a simple NamedTuple :)